src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -40,13 +41,14 @@ class remove_virtual_functionst
   const namespacet ns;
   const symbol_table_baset &symbol_table;
 
-  class_hierarchyt class_hierarchy;
+  const class_hierarchyt &class_hierarchy;
 
   void remove_virtual_function(
     goto_programt &goto_program,
     goto_programt::targett target);
-
+public:
   void get_functions(const exprt &, dispatch_table_entriest &);
+protected:
```

Comment:
I think it would be less churn and much clearer to just move the function declaration up into the `public` section.